operator/main.go

```diff
@@ -240,3 +246,38 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func setupContext(reconciler *agent.AgentReconciler) context.Context {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		<-c
+		removeFinalizer(ctx, reconciler)
+		cancel()
+	}()
+	return ctx
+}
+
+func removeFinalizer(ctx context.Context, reconciler *agent.AgentReconciler) {
+	// Remove the finalizer from the agent
+	agent := &agentv1alpha1.Agent{}
+	err := reconciler.Client.Get(ctx, types.NamespacedName{
+		Name:      os.Getenv("AGENT_CR_NAME"),
```

Comment:
This logic is restricted to the Helm chart based installation only, which is not correct.

What if someone installs operator separately and then the CR?